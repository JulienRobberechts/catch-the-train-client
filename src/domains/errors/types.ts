
export interface AError {
    errorCode: number;
    errorMessage: string;
}

export interface AnyIncomingError {
    isAxiosError?: boolean;
    response?: {
        status: number;
    } 
}

export interface GuiError {
    message1: string;
    message2: string;
    icon: string;
    colorKey: "disabled"|"normal"|"original"|"highlight"|"warning";
}

export interface AppError {
    code: number;
    msg: string;
    level?: string;
    originalException?: Error;
}

export interface PublicError {
    errorType: string;
    errorCode: number;
    errorMessage?: string;
    originalException?: Error;
}
