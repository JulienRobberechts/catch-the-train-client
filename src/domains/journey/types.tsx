export interface ReduxStateJourney {
  network: string;
  line: string;
  departure: string;
  missions: string;
  destination: string;
}

export interface Journey {
  network: string;
  line: string;
  departure: string;
  // To check...
  destination?: string;
  missions: string;
}

export interface FullJourney {
  network: string;
  line: string;
  departure: string;
  destination: string;
  missions: string;
}

export interface UserJourney {
  network: string;
  line: string;
  departure: string;
  destination: string;
}

export interface RatpJourney {
  network: string;
  line: string;
  departure: string;
  missions: string;
}
