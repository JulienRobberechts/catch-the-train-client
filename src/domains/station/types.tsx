export interface ReduxStateStation {
  userConfiguration: UserConfiguration;
  stationConfigurations: StationConfigurationMap;
}

export interface StationConfigurationMap {
  [stationSlug: string]: StationConfiguration;
}

export interface StationConfiguration {
  travelDurationSeconds: number;
  accessDurationSeconds: number;
}

export interface StationConfigurationWithStation {
  station: string;
  travelDurationSeconds: number;
  accessDurationSeconds: number;
}

export interface PayloadStationConfiguration {
  station: string;
  travelDurationSeconds: number;
  accessDurationSeconds?: number;
}

export interface UserConfiguration {
  onTimeMarginDelaySeconds: number;
  timezone?: string;
}
