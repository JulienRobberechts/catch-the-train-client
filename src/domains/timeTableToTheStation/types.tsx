import moment from "moment";
import DelayStatus from "../toTheStation/pure/delayStatus";
import { RawDeparture } from "../timeTable/types";

export interface EnhanceTimeTable {
  currentDeparture?: {
    index: number;
    code: string;
  };
  travel?: {
    nowTime: moment.Moment;
    travelDuration: moment.Duration;
    accessDuration: moment.Duration;
  };
  enhancedDepartures?: EnhancedDeparture[];
}

export interface EnhancedDeparture {
  departureIndex: number;
  trainCode: string;
  departure: RawDeparture;

  departureTime: moment.Moment;
  departureDuration: moment.Duration;

  delayDuration: moment.Duration;
  delayStatus: DelayStatus;
  onTimeMarginDelaySeconds: number;
}
