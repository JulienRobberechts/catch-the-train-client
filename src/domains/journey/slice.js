import { createSlice } from "@reduxjs/toolkit";

import { getMissions } from "./service";
import { loadJourney } from "../../adapters/journey";

export const initialState = {
  ...loadJourney(),
};

export const slice = createSlice({
  name: "journey",
  initialState,
  reducers: {
    reset: (state) => {
      Object.assign(state, initialState);
    },
    setCurrentJourney: (state, action) => {
      const { network, line, departure, destination } = action.payload;
      const missions = getMissions(departure, destination).join(",");
      state.network = network;
      state.line = line;
      state.departure = departure;
      state.destination = destination;
      state.missions = missions;
    },
  },
});

export const { reset, setCurrentJourney } = slice.actions;

export default slice.reducer;
