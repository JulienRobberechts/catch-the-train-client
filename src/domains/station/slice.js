import { createSlice } from "@reduxjs/toolkit";
import { loadStationConfigurations } from "../../adapters/stationPreferences";

export const initialState = {
  userConfiguration: {
    onTimeMarginDelaySeconds: 50,
    timezone: "+01:00",
  },
  stationConfigurations: loadStationConfigurations(),
};

export const slice = createSlice({
  name: "toTheStation",
  initialState,
  reducers: {
    reset: (state) => {
      Object.assign(state, initialState);
    },
    setUserConfiguration: (state, action) => {
      state.userConfiguration = action.payload;
    },
    setStationConfiguration: (state, action) => {
      const { station, ...props } = action.payload || {};
      if (station) {
        state.stationConfigurations[station] = { ...props };
      }
    },
  },
});

export const {
  reset,
  setStationConfiguration,
  setUserConfiguration,
} = slice.actions;

export default slice.reducer;
