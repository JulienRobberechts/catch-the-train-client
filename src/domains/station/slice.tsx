import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { loadStationConfigurations } from "../../adapters/stationPreferences";
import { PayloadStationConfiguration } from "./types";
import { ReduxStateStation } from "./types";

export const initialState: ReduxStateStation = {
  userConfiguration: {
    onTimeMarginDelaySeconds: 50,
    timezone: "+01:00",
  },
  stationConfigurations: loadStationConfigurations(),
};

export const slice = createSlice({
  name: "toTheStation",
  initialState,
  reducers: {
    reset: (state) => {
      Object.assign(state, initialState);
    },
    setUserConfiguration: (state, action) => {
      state.userConfiguration = action.payload;
    },
    setStationConfiguration: (
      state,
      action: PayloadAction<PayloadStationConfiguration>
    ) => {
      const { station, travelDurationSeconds, accessDurationSeconds } =
        action.payload || {};
      if (station) {
        if (!state.stationConfigurations[station]) {
          state.stationConfigurations[station] = {
            travelDurationSeconds: 600,
            accessDurationSeconds: 120,
          };
        }
        if (travelDurationSeconds) {
          state.stationConfigurations[
            station
          ].travelDurationSeconds = travelDurationSeconds;
        }
        if (accessDurationSeconds) {
          state.stationConfigurations[
            station
          ].accessDurationSeconds = accessDurationSeconds;
        }
      }
    },
  },
});

export const {
  reset,
  setStationConfiguration,
  setUserConfiguration,
} = slice.actions;

export default slice.reducer;
