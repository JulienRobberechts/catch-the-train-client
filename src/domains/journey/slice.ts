import { createSlice, PayloadAction } from "@reduxjs/toolkit";

import { loadJourney } from "../../adapters/journey";
import { FullJourney } from "./types";

export const initialState: FullJourney = {
  ...loadJourney(),
};

export const slice = createSlice({
  name: "journey",
  initialState,
  reducers: {
    reset: (state) => {
      Object.assign(state, initialState);
    },
    setCurrentJourney: (state, action: PayloadAction<FullJourney>) => {
      const {
        network,
        line,
        departure,
        destination,
        missions,
      } = action.payload;
      state.network = network;
      state.line = line;
      state.departure = departure;
      state.destination = destination;
      state.missions = missions;
    },
  },
});

export const { reset, setCurrentJourney } = slice.actions;

export default slice.reducer;
