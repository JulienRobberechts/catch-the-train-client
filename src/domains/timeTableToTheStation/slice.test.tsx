import { initialState as timeTableInitialState } from "../timeTable/slice";
import { initialState as toTheStationInitialState } from "../toTheStation/slice";
import {
  selectEnhancedTimeTable,
  selectCurrentStationConfiguration,
} from "./selectors";
import { RootState } from "../../redux-store/types";

const sampleDepartures = [
  {
    trainCode: "0824",
    departureTime: "2020-03-10T08:24:00.000Z",
    mission: "UPAC",
    displayAttributes: "08:24",
    displayDestination: "Cergy-Le-Haut",
  },
  {
    trainCode: "0829",
    departureTime: "2020-03-10T08:29:00.000Z",
    mission: "UPAC",
    displayAttributes: "08:29",
    displayDestination: "Cergy-Le-Haut",
  },
];

describe("slice timeTable", () => {
  const initialRootState: RootState = {
    timeTable: timeTableInitialState,
    toTheStation: toTheStationInitialState,
  };
  describe("initial state", () => {
    it("'selectEnhancedTimeTable' should return falsy", () => {
      expect(selectEnhancedTimeTable(initialRootState)).toEqual({ });
    });
  });
  describe("after initialization", () => {
    const sampleUserConfiguration = {
      onTimeMarginDelaySeconds: 22,
      timezone: "+04:00",
    };
    const sampleStationConfigurations = {
      "chatelet+les+halles": {
        travelDurationSeconds: 555,
        accessDurationSeconds: 77,
      },
      "other-station": {
        travelDurationSeconds: 1000,
        accessDurationSeconds: 30,
      },
    };
    const samplejourneyRerAChatelet = {
      network: "rers",
      line: "a",
      departure: "chatelet+les+halles",
      destination: "auber",
      missions: "ZEBU",
    };

    const rootState: RootState = {
      journey: samplejourneyRerAChatelet,
      station: {
        userConfiguration: sampleUserConfiguration,
        stationConfigurations: sampleStationConfigurations,
      },
      toTheStation: {
        currentTime: "2020-03-10T08:13:30.000Z",
        currentTrainCode: "0824",
      },
      timeTable: {
        loading: false,
        error: null,
        data: {
          departures: sampleDepartures,
        },
      },
    };
    it("'selectEnhancedTimeTable' should MatchSnapshot", () => {
      const actualEnhancedTimeTable = selectEnhancedTimeTable(rootState);
      expect(actualEnhancedTimeTable).toBeTruthy();

      const {
        currentDeparture,
        enhancedDepartures,
        travel,
      } = actualEnhancedTimeTable;

      expect(currentDeparture).toBeTruthy();
      expect(currentDeparture?.code).toBe("0824");
      expect(currentDeparture?.index).toBe(0);

      expect(enhancedDepartures).toBeTruthy();
      const enhancedDepartures2 = enhancedDepartures!;
      expect(enhancedDepartures2.length).toBe(2);

      expect(enhancedDepartures2[0].departureIndex).toBe(0);
      expect(enhancedDepartures2[0].trainCode).toBe("0824");

      expect(enhancedDepartures2[0].departureTime.toISOString()).toBe(
        "2020-03-10T08:24:00.000Z"
      );
      expect(enhancedDepartures2[0].delayDuration.asSeconds()).toBe(-2); // 630 - 555 - 77 = -2
      expect(enhancedDepartures2[0].delayStatus).toBe("on-time");
      expect(enhancedDepartures2[0].onTimeMarginDelaySeconds).toBe(22);

      expect(travel).toBeTruthy();
      expect(travel?.accessDuration.asSeconds()).toBe(77);
      expect(travel?.nowTime.toISOString()).toBe("2020-03-10T08:13:30.000Z");
      expect(travel?.travelDuration.asSeconds()).toBe(555);

      expect(actualEnhancedTimeTable).toMatchSnapshot();
    });
    it("'selectCurrentStationConfiguration' should MatchSnapshot", () => {
      const actualCurrentStationConfiguration = selectCurrentStationConfiguration(
        rootState
      );
      expect(actualCurrentStationConfiguration).toBeTruthy();
      expect(actualCurrentStationConfiguration?.travelDurationSeconds).toBe(555);
      expect(actualCurrentStationConfiguration?.accessDurationSeconds).toBe(77);
    });
  });
});
