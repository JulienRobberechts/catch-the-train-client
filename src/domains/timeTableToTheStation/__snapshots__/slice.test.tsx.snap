// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`slice timeTable after initialization 'selectEnhancedTimeTable' should MatchSnapshot 1`] = `
Object {
  "currentDeparture": Object {
    "code": "0824",
    "index": 0,
  },
  "enhancedDepartures": Array [
    Object {
      "delayDuration": "-PT2S",
      "delayStatus": "on-time",
      "departure": Object {
        "departureTime": "2020-03-10T08:24:00.000Z",
        "displayAttributes": "08:24",
        "displayDestination": "Cergy-Le-Haut",
        "trainCode": "0824",
      },
      "departureDuration": "PT10M30S",
      "departureIndex": 0,
      "departureTime": "2020-03-10T08:24:00.000Z",
      "onTimeMarginDelaySeconds": 22,
      "trainCode": "0824",
    },
    Object {
      "delayDuration": "PT4M58S",
      "delayStatus": "early",
      "departure": Object {
        "departureTime": "2020-03-10T08:29:00.000Z",
        "displayAttributes": "08:29",
        "displayDestination": "Cergy-Le-Haut",
        "trainCode": "0829",
      },
      "departureDuration": "PT15M30S",
      "departureIndex": 1,
      "departureTime": "2020-03-10T08:29:00.000Z",
      "onTimeMarginDelaySeconds": 22,
      "trainCode": "0829",
    },
  ],
  "travel": Object {
    "accessDuration": "PT1M17S",
    "nowTime": "2020-03-10T08:13:30.000Z",
    "travelDuration": "PT9M15S",
  },
}
`;
